### PR TITLE
Add more tests to LandGrabInSpaceTests.cs

### PR DESCRIPTION
**Pull request by cizra**
See discussion at https://forum.exercism.org/t/land-grab-in-space-has-insufficient-test-coverage-for-the-plot-with-longest-side/9615

### CODE REVIEW

File: exercises/concept/land-grab-in-space/LandGrabInSpaceTests.cs [hunk 1]
```diff
@@ -59,6 +59,32 @@ public void GetLongestSide()
...
+    }
+
+    [Fact]
+    [Task(4)]
+    public void GetLongestSideDiagonal1()
+    {
+        var ch = new ClaimsHandler();
+        // Bottom left, bottom right, top left, top right. The longest side is 10, diagonal is >14
+        var shorter = CreatePlot(new Coord(0, 0), new Coord(0, 10), new Coord(10, 0), new Coord(10, 10));
+        // Bottom left, top left, top right, bottom right, longest side 11
```
Comment (ErikSchierboom) [inline exercises/concept/land-grab-in-space/LandGrabInSpaceTests.cs:81]: Not sure about the comments. Does the student need them?

```suggestion
        var shorter = CreatePlot(new Coord(0, 0), new Coord(0, 10), new Coord(10, 0), new Coord(10, 10));
```

**Comment by ErikSchierboom** [comment]
Thanks!

**Comment by Naeddyr** [comment]
Adding these tests made the "learning exercise" significantly harder. A better solution would be to add a guarantee that the coordinates of the plot are always in the correct order.

**Comment by cizra** [comment]
> Adding these tests made the "learning exercise" significantly harder. A better solution would be to add a guarantee that the coordinates of the plot are always in the correct order.

Disagreed, @Naeddyr - before adding these tests, the most difficult task in the exercise could be completed by just returning the 1st element of the list, which is not in the spirit of the task. The task was about calculating sizes of plots, and this just wasn't necessary. To be honest, the bit with dialog could have been skipped, tho.

**Comment by Naeddyr** [comment]
The diagonals test makes this learning exercise (originally about hash sets and hashing and marginally about the data itself) an intermediate exercise about generating a convex polygon from a set of points or coming up with some other algorithm. You shouldn't just add a completely new category of test because it feels right. The correct way to do this would have been to make the old tests better. Randomize the elements of the list, add more elements to the list, that sort of thing.

**Pull request merged**